Sources/SwiftFrameCore/Workers/TextRenderer.swift

```diff
@@ -0,0 +1,100 @@
+import AppKit
+import Foundation
+
+final class TextRenderer {
+
+    private let ky_MinFontSize: CGFloat = 1
+    private let ky_pointSizeTolerance: CGFloat = 1e-7
+
+    // MARK: - Frame Rendering
+
+    func render(text: String, font: NSFont, color: NSColor, alignment: TextAlignment, rect: NSRect, context: CGContext) throws {
+        let attributedString = try makeAttributedString(for: text, font: font, color: color, alignment: alignment)
+
+        context.saveGState()
+        defer { context.restoreGState() }
```

Comment:
Nit: Similar to the other case. This works fine. For me personally it just creates this invisible `goto` at the end of the method body and it seems not really needed since the body is so simple. But might be just me. :)